app/lib/actions/katello/repository/finish_upload.rb

```diff
@@ -5,7 +5,9 @@ class FinishUpload < Actions::Base
         def plan(repository, options = {})
           dependency = options.fetch(:dependency, nil)
           generate_metadata = options.fetch(:generate_metadata, true)
+          sync_capsule = options.fetch(:sync_capsule, true)
           plan_action(Katello::Repository::MetadataGenerate, repository, :dependency => dependency) if generate_metadata
+          plan_action(Katello::Repository::CapsuleGenerateAndSync, repository) if sync_capsule
```

Comment:
So doing it this will cause the upload task to actually not finish until all the capsules are done syncing.  This is normally how we do capsule syncs, instead we usually wait until the run phase to start that process: 

https://github.com/Katello/katello/blob/master/app/lib/actions/katello/repository/sync.rb#L49